src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/TaskResourceSpecification.java

```diff
@@ -0,0 +1,161 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.cft.query;
+
+import org.springframework.data.jpa.domain.Specification;
+import uk.gov.hmcts.reform.wataskmanagementapi.auth.access.entities.AccessControlResponse;
+import uk.gov.hmcts.reform.wataskmanagementapi.auth.permission.entities.PermissionTypes;
+import uk.gov.hmcts.reform.wataskmanagementapi.auth.role.entities.RoleAssignment;
+import uk.gov.hmcts.reform.wataskmanagementapi.auth.role.entities.enums.Classification;
+import uk.gov.hmcts.reform.wataskmanagementapi.cft.entities.TaskResource;
+import uk.gov.hmcts.reform.wataskmanagementapi.cft.entities.TaskRoleResource;
+import uk.gov.hmcts.reform.wataskmanagementapi.cft.enums.CFTTaskState;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.SearchTaskRequest;
+import uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.camunda.SecurityClassification;
+import uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.search.SearchParameter;
+import uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.search.SearchParameterKey;
+
+import java.util.ArrayList;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+import javax.persistence.criteria.Join;
+import javax.persistence.criteria.Predicate;
+
+@SuppressWarnings({"PMD.DataflowAnomalyAnalysis"})
+public final class TaskResourceSpecification {
+
+    private TaskResourceSpecification(){
+        // avoid creating object
+    }
+
+    public static Specification<TaskResource> getTasks(
+            SearchTaskRequest searchTaskRequest,
+            AccessControlResponse accessControlResponse,
+            List<PermissionTypes> permissionsRequired
+    ) {
+
+        return searchByRoles(permissionsRequired, accessControlResponse)
+            .and(searchByClassification(accessControlResponse)
+            .and(searchByRegion()
+            .and(searchByState(searchTaskRequest)
+            .and(searchByLocation(searchTaskRequest)
+            .and(searchByJurisdiction(searchTaskRequest)
+            .and(searchByCaseId(searchTaskRequest)
+            .and(searchByUser(searchTaskRequest))))))));
+    }
+
+    public static Specification<TaskResource> searchByRoles(List<PermissionTypes> permissionsRequired,
```

Comment:
do these have to be static?

Reply:
The reason for using static is, The Specification interface itself has static methods like and, or, where..etc. So followed the same way of implementing filters. Moreover, the filters does not hold any state and don't see a benefit of creating instances to access those methods.

Reply:
Ok since these sounds more like a helper class to create a specification with no instance variables, sounds like static methods are ok.